query_result.go

```diff
@@ -204,11 +204,10 @@ func (qr *QueryResult) parseEdge(cell interface{}) *Edge {
 func (qr *QueryResult) parseArray(cell interface{}) []interface{} {
 	var array = cell.([]interface{})
 	var arrayLength = len(array)
-	var res = make([]interface{}, arrayLength)
 	for i := 0; i < arrayLength; i++ {
-		res[i] = qr.parseScalar(array[i].([]interface{}))
+		array[i] = qr.parseScalar(array[i].([]interface{}))
```

Comment:
Should the cast be to an interface{}
currently we're treating array[i] as another array.